multicluster/controllers/multicluster/label_identity_controller.go

```diff
@@ -0,0 +1,305 @@
+	}
+	if ok && hostNetwork {
+		// Non-HostNetwork Pod update to a HostNetwork Pod.
+		return "", deleteLabel(podKey, oldNormalizedLabel)
+	}
+	// Non-HostNetwork Pod update.
+	if normalizedLabel == oldNormalizedLabel {
+		return "", ""
+	}
+	labelToDelete = deleteLabel(podKey, oldNormalizedLabel)
+	labelToAdd = addLabel(podKey, normalizedLabel)
+	return
+}
+
+func (r *LabelIdentityReconciler) addLabelIdentityResourceExport(ctx context.Context, normalizedLabel string) error {
+	r.resourceExportMutex.Lock()
+	defer r.resourceExportMutex.Unlock()
+
+	existingResExport := &mcsv1alpha1.ResourceExport{}
+	resNamespaced := types.NamespacedName{
+		Name:      getResourceExportNameForLabelIdentity(r.localClusterID, normalizedLabel),
+		Namespace: r.remoteCommonArea.GetNamespace(),
+	}
+	err := r.remoteCommonArea.Get(ctx, resNamespaced, existingResExport)
+	if err != nil && !apierrors.IsNotFound(err) {
+		klog.ErrorS(err, "Failed to get ResourceExport in remote cluster", "resourceexport", resNamespaced)
+		return err
+	}
+	if apierrors.IsNotFound(err) {
+		labelResExport := r.getLabelIdentityResourceExport(r.remoteCommonArea.GetNamespace(), resNamespaced.Name, normalizedLabel)
+		if err = r.remoteCommonArea.Create(ctx, labelResExport, &client.CreateOptions{}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *LabelIdentityReconciler) deleteLabelIdentityResourceExport(ctx context.Context, normalizedLabel string) error {
+	r.resourceExportMutex.Lock()
+	defer r.resourceExportMutex.Unlock()
+	labelResExport := r.getLabelIdentityResourceExport(r.remoteCommonArea.GetNamespace(), getResourceExportNameForLabelIdentity(r.localClusterID, normalizedLabel), normalizedLabel)
+	if err := r.remoteCommonArea.Delete(ctx, labelResExport, &client.DeleteOptions{}); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+	return nil
+}
+
+func (r *LabelIdentityReconciler) getLabelIdentityResourceExport(resExportNamespace, name string, normalizedLabel string) *mcsv1alpha1.ResourceExport {
+	return &mcsv1alpha1.ResourceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: resExportNamespace,
+			Labels: map[string]string{
+				common.SourceKind:      common.LabelIdentityKind,
+				common.SourceClusterID: r.localClusterID,
+			},
+		},
+		Spec: mcsv1alpha1.ResourceExportSpec{
+			ClusterID: r.localClusterID,
+			Kind:      common.LabelIdentityKind,
+			LabelIdentities: &mcsv1alpha1.LabelIdentityExport{
+				NormalizedLabel: normalizedLabel,
+			},
+		},
+	}
+}
+
+func getNormalizedLabel(nsLabels, podLabels map[string]string) string {
+	return "namespace:" + labels.FormatLabels(nsLabels) + "&pod:" + labels.FormatLabels(podLabels)
+}
+
+// getResourceExportNameForLabelIdentity retrieves the ResourceExport name for exporting
+// label identities in that cluster.
+func getResourceExportNameForLabelIdentity(clusterID string, normalizedLabel string) string {
+	return clusterID + "-" + normalizedLabel
```

Comment:
That's a good point. How do you think this solution:
Use `clusterID + "-" + hashLabelIdentity(normalizedLabel)` as this ResourceExport's name. This could also fix the issue that labels also have "-". But this will require `LabelIdentityExportReconciler` to cache another map from `hashValue` to `normalizedLabel` in order to handle delete event.

Reply:
Just realized this is what `LabelIdentityImportReconciler` do. Will update.

Reply:
Updated. Now the ResourceExport name uses the label identity hash. Some other refactors are done to accommodate this change